web_src/js/features/comp/EditorMarkdown.ts

```diff
@@ -4,6 +4,16 @@ export function triggerEditorContentChanged(target) {
   target.dispatchEvent(new CustomEvent(EventEditorContentChanged, {bubbles: true}));
 }
 
+export function textareaInsertText(textarea, value) {
+  const startPos = textarea.selectionStart;
+  const endPos = textarea.selectionEnd;
+  textarea.value = textarea.value.substring(0, startPos) + value + textarea.value.substring(endPos);
+  textarea.selectionStart = startPos;
+  textarea.selectionEnd = startPos + value.length;
+  textarea.focus();
+  triggerEditorContentChanged(textarea);
+}
```

Comment:
Can it use https://github.com/go-gitea/gitea/blob/main/web_src/js/utils/dom.ts#L277?

That method has the benefit that CTRL-Z works after it.

Reply:
No, it can't at the moment.

I know the existence of replaceTextareaSelection but it needs some extra work to merge these two.

Reply:
The only difference seems to be that this function re-selects the inserted text and focuses the textarea. The re-selection could be pictured in a option and the focus could be done after the function call.

```
replaceTextareaSelection(textarea: HTMLTextAreaElement, text: string, {reselect: false}: {reselect: boolean})

Reply:
> `triggerEditorContentChanged` does the same as a `change` event, right?

No.

----

I do not doing more changes is in this PR's scope since it has been fully tested.

And this `textareaInsertText` is not a new function, it is just extracted from `insertPlaceholder`, nothing worse.

Reply:
I'd prefer if this at least does not break the textarea history. Does CTRL-Z work after inserting content?

Reply:
This textareaInsertText is not a new function, it is just extracted from insertPlaceholder, nothing worse.

There is always a chance to propose new improvements.

Reply:
Should fix it later.